src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs

```diff
@@ -653,12 +654,21 @@ await _projectSystemProjectFactory.ApplyBatchChangeToWorkspaceMaybeAsync(useAsyn
                         newSolution: solutionChanges.Solution.RemoveProjectReference(Id, projectReference));
                 }
 
+                // Analyzer reference removing...
+                {
+                    projectUpdateState = projectUpdateState.WithIncrementalAnalyzerReferencesRemoved(_analyzersRemovedInBatch);
+
+                    foreach (var analyzerReference in _analyzersRemovedInBatch)
+                        solutionChanges.UpdateSolutionForProjectAction(Id, solutionChanges.Solution.RemoveAnalyzerReference(Id, analyzerReference));
```

Comment:
doesn't look exactly the same as below as we don't hav ea Solution.RemoveAnalyzerReferences (plural) to go with Solution.AddAnalyzerReferences.  And i'm not interested in adding that at this point :)